ReactAndroid/src/androidTest/js/UIManagerTestModule.js

```diff
@@ -9,214 +9,248 @@
-      width: 50,
-      height: 60,
-    },
-  }),
-  render: function() {
+type AbsolutePositionBottomRightTestAppProps = $ReadOnly<{||}>;
+class AbsolutePositionBottomRightTestApp extends React.Component<
+  AbsolutePositionBottomRightTestAppProps,
+> {
+  render() {
     return (
       <View
-        style={this._styles.container}
+        style={AbsolutePositionBottomRightTestAppStyles.container}
         testID="container"
         collapsable={false}>
-        <View style={this._styles.absolute} collapsable={false} />
+        <View
+          style={AbsolutePositionBottomRightTestAppStyles.absolute}
+          collapsable={false}
+        />
       </View>
     );
+  }
+}
+
+const AbsolutePositionBottomRightTestAppStyles = StyleSheet.create({
+  container: {
+    width: 100,
+    height: 100,
+  },
+  absolute: {
+    position: 'absolute',
+    bottom: 15,
+    right: 10,
+    width: 50,
+    height: 60,
   },
 });
 
-var CenteredTextView = createReactClass({
-  displayName: 'CenteredTextView',
-  _styles: StyleSheet.create({
-    parent: {
-      width: 200,
-      height: 100,
-      backgroundColor: '#aa3311',
-      justifyContent: 'center',
-      alignItems: 'center',
-    },
-    text: {
-      fontSize: 15,
-      color: '#672831',
-    },
-  }),
-  render: function() {
+type CenteredTextViewProps = $ReadOnly<{|
+  text?: ?string,
+|}>;
+class CenteredTextView extends React.Component<CenteredTextViewProps> {
+  render() {
     return (
       <View collapsable={false}>
-        <View style={this._styles.parent} collapsable={false}>
-          <Text style={this._styles.text} testID="text">
+        <View style={CenteredTextViewStyles.parent} collapsable={false}>
+          <Text style={CenteredTextViewStyles.text} testID="text">
             {this.props.text}
           </Text>
         </View>
       </View>
     );
+  }
+}
+
+const CenteredTextViewStyles = StyleSheet.create({
+  parent: {
+    width: 200,
+    height: 100,
+    backgroundColor: '#aa3311',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  text: {
+    fontSize: 15,
+    color: '#672831',
   },
 });
 
-var flushUpdatePositionInList = null;
-var UpdatePositionInListTestApp = createReactClass({
-  displayName: 'UpdatePositionInListTestApp',
-  _styles: StyleSheet.create({
-    element: {
-      height: 10,
-    },
-    active: {
-      height: 50,
-    },
-  }),
-  getInitialState: function() {
-    flushUpdatePositionInList = () => this.setState({active: true});
-    return {active: false};
-  },
-  render: function() {
+let flushUpdatePositionInList = null;
+type UpdatePositionInListTestAppProps = $ReadOnly<{||}>;
+type UpdatePositionInListTestAppState = $ReadOnly<{|
+  active?: ?boolean,
+|}>;
+class UpdatePositionInListTestApp extends React.Component<
+  UpdatePositionInListTestAppProps,
+  UpdatePositionInListTestAppState,
+> {
+  state = {
+    active: false,
+  };
+
+  flushUpdatePositionInList = () => this.setState({active: true});
+
```

Comment:
This seems like a mistake. 

Right now, this is creating a property on `UpdatePositionInListTestApp` instances called `flushUpdatePositionInList`, which isn't what the original code does. What we actually need is something like this:

```JavaScript
class UpdatePositionInListTestApp extends React.Component<
  UpdatePositionInListTestAppProps,
  UpdatePositionInListTestAppState,
> {
  state = {
    active: false,
  }

  constructor(...args) {
    super(...args);
    flushUpdatePositionInList = () => this.setState({active: true});
  }
```

Reply:
Thanks, for your feedback, fixed.

Reply:
@RSNara It seemed a bit confusing to me that the instance of this class is being created [here,](https://github.com/facebook/react-native/blob/master/ReactAndroid/src/androidTest/java/com/facebook/react/tests/CatalystUIManagerTestCase.java#L176), I believe that we do not need method inside the constructor.


```
  public void testUpdatePositionInList() {
    FrameLayout rootView = createRootView();
    jsModule.renderUpdatePositionInListTestApplication(rootView.getId());
    waitForBridgeAndUIIdle();
...

    // Let's make the second elements 50px height instead of 10px
    jsModule.flushUpdatePositionInList();
    waitForBridgeAndUIIdle();
....
  }

```
And now I'm not sure if we need this 
```
  flushUpdatePositionInList() {
    flushUpdatePositionInList();
  },
```
Let me know what do you think, to help me.

Reply:
I'm afraid I don't understand where you're confused. 

That test looks like it renders the component, makes a few assertions, calls `flushUpdatePositionInList`, which updates the state and re-renders the component, and then makes a few more assertions. 

We need to expose `flushUpdatePositionInList` because we need a way to re-render the component with new state after we make the first batch of assertions. 

We can't call `this.flushUpdatePositionInList();` inside the constructor, because (1) `setState` doesn't work inside the constructor and (2), even if it did work, we don't want to immediately re-render the component with the new state. Doing so would make the first batch of assertions fail in the test. 

I think there may have been some sort of miscommunication earlier. Let's change the code to this:

```JavaScript
let flushUpdatePositionInList = null;

// ...
class UpdatePositionInListTestApp extends React.Component<
  UpdatePositionInListTestAppProps,
  UpdatePositionInListTestAppState,
> {
  state = {
    active: false,
  }

  constructor(...args) {
    super(...args);
    flushUpdatePositionInList = () => this.setState({active: true});
  }
  // ...
}

const UITestManagerModule = {
  // ...
  flushUpdatePositionInList,
}
```

Reply:
Thanks a lot, awesome